app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -191,24 +198,21 @@ class LuciBuildService {
     );
   }
 
-  /// Schedules presubmit [builders] on BuildBucket for [pullRequest].
-  Future<List<String>> scheduleTryBuilds({
-    required List<LuciBuilder> builders,
+  /// Schedules presubmit [targets] on BuildBucket for [pullRequest].
+  Future<List<Target>> scheduleTryBuilds({
+    required List<Target> targets,
     required github.PullRequest pullRequest,
     CheckSuiteEvent? checkSuiteEvent,
   }) async {
     if (!config.githubPresubmitSupportedRepo(pullRequest.base!.repo!.slug())) {
-      throw BadRequestException('Repository ${pullRequest.base!.repo!.fullName} is not supported by this service.');
+      throw BadRequestException('${pullRequest.base!.repo!.slug()} is not supported by this service.');
     }
-    final List<String> builderNames = await _builderNamesForRepo(
-      builders,
-      pullRequest,
-    );
+    targets = await _targetsNotScheduled(targets, pullRequest);
     int retryCount = 1;
-    List<String> remainingBuildersToSchedule = List<String>.from(builderNames);
-    while (builderNames.isNotEmpty) {
-      remainingBuildersToSchedule = await _scheduleTryBuilds(
-        builderNames: remainingBuildersToSchedule,
+    List<Target> remainingTargetsToSchedule = List<Target>.from(targets);
+    while (remainingTargetsToSchedule.isNotEmpty) {
```

Comment:
Related to the another comment: is there any way we can catch those queued builds here, so that a retry is triggered for them?

Reply:
I believe you're referring to something like https://github.com/flutter/flutter/issues/86948

In that case, the issue is we failed to schedule the build entirely. So it wouldn't show in the running list from LUCI, and it would be scheduled here (as expected)